### DISCUSSION TITLE
Phone number provider

### DISCUSSION THREAD

**Discussion by josearomeroj** [go-pkgz/auth#146] [Ideas]
Before start working in the pull request...

What do you think about adding a phone provider using the the google identity toolkit?

Ref: 
- [sendVerificationCode](https://cloud.google.com/identity-platform/docs/reference/rest/v1/accounts/sendVerificationCode)
- [signInWithPhoneNumber](https://cloud.google.com/identity-platform/docs/reference/rest/v1/accounts/signInWithPhoneNumber)

It would need a captcha implementation from the frontend.

**Comment by umputun** [comment]
an interesting one. I guess it will be some mix between the regular oauth2 flow plus a confirmation like we do for email (verify) provider.
